addons/mail/controllers/mail_template.py

```diff
@@ -0,0 +1,24 @@
+from odoo import http
+from odoo.http import request
+from odoo.osv import expression
+
+
+class MailTemplateController(http.Controller):
+    @http.route('/mail/template', type='json', auth='user')
+    def fetch_mail_templates(self, model, limit=10):
+        Template = request.env['mail.template']
+        fields = ['id', 'display_name']
+        domain = [('model', '=', model)]
+
+        readable_templates = Template.search_read(
+            domain, fields, limit=limit)
+        writable_templates = Template.search_read(
+            expression.AND([domain, [('can_write', '=', True)]]), fields, limit=limit)
+        unlinkable_templates = Template.search_read(
+            expression.AND([domain, [('can_unlink', '=', True)]]), fields, limit=limit)
+
+        return {
+            'readable_templates': readable_templates,
+            'writable_templates': writable_templates,
+            'unlinkable_templates': unlinkable_templates
+        }
```

Comment:
Would say templates naming is redundant
```suggestion
        return {
            'readable': readable_templates,
            'writable': writable_templates,
            'unlinkable': unlinkable_templates
        }
```